shell/browser/api/electron_api_web_contents.cc

```diff
@@ -3708,7 +3708,13 @@ gin::Handle<WebContents> WebContents::New(
     const gin_helper::Dictionary& options) {
   gin::Handle<WebContents> handle =
       gin::CreateHandle(isolate, new WebContents(isolate, options));
+  // Send unhandled errors within init to a different scope,
+  // so they're handled within JavaScript and don't crash within C++
+  v8::TryCatch try_catch(isolate);
   gin_helper::CallMethod(isolate, handle.get(), "_init");
+  if (try_catch.HasCaught()) {
+    node::errors::TriggerUncaughtException(isolate, try_catch);
+  }
```

Comment:
It might be useful to have a helper method that does this logic? eg. `gin_helper::CallMethodCatchException` or something

Reply:
@nornagon Great idea, changed it add and use the helper 🙇‍♀️